addons/token-exchange/secret_exchange_handler_register.go

```diff
@@ -0,0 +1,43 @@
+package addons
+
+import (
+	"fmt"
+
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	"k8s.io/client-go/rest"
+)
+
+type SecretExchangeHandler struct {
+	RegisteredHandlers map[string]SecretExchangeHandlerInerface
+}
+
+var secretExchangeHandler *SecretExchangeHandler
+
+// intialize secretExchangeHandler with handlers
+func registerHandler(spokeKubeConfig *rest.Config) error {
+	if secretExchangeHandler == nil {
+
+		// rook specific client
+		rookClient, err := rookclient.NewForConfig(spokeKubeConfig)
+		if err != nil {
+			return fmt.Errorf("failed to add rook client: %v", err)
+		}
+
+		// a generic spoke client which is common between all handlers
+		genericSpokeClient, err := getClient(spokeKubeConfig)
+		if err != nil {
+			return err
+		}
+
+		secretExchangeHandler = &SecretExchangeHandler{
+			RegisteredHandlers: map[string]SecretExchangeHandlerInerface{
+				RookSecretHandlerName: rookSecretHandler{
+					spokeClient: genericSpokeClient,
+					rookClient:  rookClient,
```

Comment:
the client which i am creating from getClient is accessable by all handlers, no need to call getClient function everytime.

whenever new handler is created all they need to do is, just  create handler strcut with client and pass genericSpokeClient.  so all struct function inside handlers  have this client.

Also, writting unit testing is easy. I can easily mock the client.